### ISSUE TITLE
Unused asset generation under specific conditions.

### ISSUE THREAD

**Issue by VimYoung** [slint-ui/slint#10728]
### Bug Description

There are warning of unused static globals under certain circumstances. Ideally, as stated in the discussion, if something is not used, it shouldn't be generated by the macro. The unused variable looks like following (if that matters):
```rust
    static SLINT_EMBEDDED_RESOURCE_2: &'static [u8] = b"<svg xmlns=\"http://www.w3.org/2000/svg\" width=\"24\" height=\"24\" viewBox=\"0 0 24 24\" fill=\"none\" stroke=\"currentColor\" stroke-width=\"2\" stroke-linecap=\"round\" stroke-linejoin=\"round\" class=\"feather feather-volume-x\"><polygon points=\"11 5 6 9 2 9 2 15 6 15 11 19 11 5\"></polygon><line x1=\"23\" y1=\"9\" x2=\"17\" y2=\"15\"></line><line x1=\"17\" y1=\"9\" x2=\"23\" y2=\"15\"></line></svg>";
```
Link of discussion: [ref](https://chat.slint.dev/public/pl/6er9sqakop833dkmqnbz6hsu5o).

### Reproducible Code (if applicable)

```slint
// Clone slint-rust-template repository and add the following contents at the bottom of `app-window.slint`.
export component GeneralSlider {
    property <bool> enable: true;
    property <int> value: 50;
    in property <image> icon: @image-url("../assets/icons/volume.svg");
    in property <image> icon-closed: @image-url("../assets/icons/mute.svg");

    HorizontalLayout {
        spacing: 2px;
        Rectangle {
            Image {
                colorize: red;
                source: enable ? icon : icon-closed;
            }
        }
    }
}

export component InputEx inherits Window {
    GeneralSlider { }
}
// Remember that InputEx is never used. This code will also generate some no code
// generation warnings which can be avoided by placing GeneralSlider in a separate file.
```

### Environment Details

- Slint Version: 1.14.1
- Platform/OS: Linux (6.18.8-3-cachyos)
- Programming Language: rust
- Backend/Renderer: Default renderer in case of Linux (probably qt).

**Comment by ogoffart** [comment]
Thanks for filing an issue.

The problem is that we collect the assets before some optimization like const propagation and inlining on our LLR datastructures. Then some unused assets remain.

We probably need some refactoring to keep better track on the assets that would allow us, for example, to also optimize things like `source.width` at compile time if the image is known, as well as removing assets no longer used.

Either that or we just add a `#[allow(dead_code)]` as a workaround.
